presto-mixin/dashboards/presto-overview.libsonnet

```diff
@@ -0,0 +1,962 @@
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        insertNulls: false,
+        lineInterpolation: 'smooth',
+        lineWidth: 2,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'never',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: 'none',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'bottom',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local queuedQueriesPanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'com_facebook_presto_execution_QueryManager_QueuedQueries{job=~"$job", presto_cluster=~"$presto_cluster"}',
+      datasource=promDatasource,
+      legendFormat='{{presto_cluster}}',
+      format='time_series',
+    ),
+  ],
+  type: 'timeseries',
+  title: 'Queued queries',
+  description: 'The number of queued queries.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'palette-classic',
+      },
+      custom: {
+        axisBorderShow: false,
+        axisCenteredZero: false,
+        axisColorMode: 'text',
+        axisLabel: '',
+        axisPlacement: 'auto',
+        barAlignment: 0,
+        drawStyle: 'bars',
+        fillOpacity: 20,
+        gradientMode: 'none',
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        insertNulls: false,
+        lineInterpolation: 'smooth',
+        lineWidth: 2,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'never',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: 'none',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'bottom',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local userErrorFailuresOneMinuteRatePanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'com_facebook_presto_execution_QueryManager_UserErrorFailures_OneMinute_Rate{job=~"$job", presto_cluster=~"$presto_cluster"}',
```

Comment:
Could you give us an update in regards to any difference between results from pre calculate rate against Grafana calculating it?

Reply:
Yea, it seems they are about the same. Using rate from Grafana combined with totalCount metric shows this
<img width="1418" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/6fab173b-d110-45a8-ab14-03f1618fcec3">
where the first two peaks are 2.22, Keeping in mind this uses rate_interval.
Using the precalculated rate shows this
<img width="1845" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/20f9122e-2b46-4611-bb31-046d8187f9f9">
excluding the first peak that started  at 14:25, the next two peaks are 6.43 and 3.65 respectively, the sum of these two would be 10.08, divide it by four since rate interval of the first is 4x the one minute, the answer is around 2.52. which is close to the peaks of the 2.22 each. I suspect the variance comes from the reference of when the calculations start. I also noticed Data dogs using the One_Minute version of the metrics, which i assume is to save on the calculations.

Reply:
Cool that's good, let's move forward with the pre calculated metrics.